### PR TITLE
Add setup_dev

### PR DESCRIPTION
**Pull request by vegaro**
Added a `setup_dev` lane to fastlane that setups the pre-commit hook that will run detekt before commiting.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ We use sdkman to specify the environment that should be used when developing. In
 [official instructions](https://sdkman.io/install). Navigate to the root of this project and do `sdk env install`,
 which should install the specific Java version we expect you to use.
 
+```bash
+$fastlane setup_dev
```
Comment (tonidero) [inline CONTRIBUTING.md:22]: Should this be `bundle exec fastlane setup_dev`?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -38,6 +38,19 @@ platform :android do
...
   end
 
+  desc "Setup development environment"
+  lane :setup_dev do |options|
+    pre_commit_link = "../.git/hooks/pre-commit"
+    if File.exist?(pre_commit_link) || File.symlink?(pre_commit_link)
+      UI.message("pre-commit script already linked ✅")
+    else
+      UI.message("Linking pre-commit script 🔗")
+      Dir.chdir ".." do
```
Comment (tonidero) [inline fastlane/Fastfile:48]: Nitpick but I prefer to use parenthesis for these 
```suggestion
      Dir.chdir("..") do
```

File: scripts/pre-commit.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+echo "Running detekt check..."
+OUTPUT="/tmp/detekt-$(date +%s)"
+./gradlew detektAll > $OUTPUT
```
Comment (tonidero) [inline scripts/pre-commit.sh:4]: So I guess this will only work when run from the root folder right? I think it's probably ok, though.

Comment (tonidero) [inline scripts/pre-commit.sh:4]: Also, I'm wondering, should this be using `&>`? That would redirect stderr to the file as well... I don't remember if gradle outputs errors there...
```suggestion
./gradlew detektAll &> $OUTPUT
```

**Review by tonidero — APPROVED** [review]
Nice job adding this! Will be very useful :)

File: scripts/pre-commit.sh (hunk 3 above)
Comment (vegaro) [reply scripts/pre-commit.sh:4]: I do see the errors when trying to commit something that detekt complains about so I think this is fine

Comment (vegaro) [reply scripts/pre-commit.sh:4]: Git will change the current working dir when this script executes so I think it's fine:

> Before Git invokes a hook, it changes its working directory to either $GIT_DIR in a bare repository or the root of the working tree in a non-bare repository. An exception are hooks triggered during a push (pre-receive, update, post-receive, post-update, push-to-checkout) which are always executed in $GIT_DIR.

https://git-scm.com/docs/githooks

**Pull request merged**